### PR TITLE
Feature/booking form

### PR DESCRIPTION
**Pull request by sepehrnoor**
## Explain the changes you’ve made

I have added the ability to book a meeting through the app, as well as retrieving existing meetings from the backend.

## Explain why these changes are made

These changes are necessary in order to give the user of the app the ability to book a meeting and see their existing meetings.

## Explain your solution

In order to implement this feature, I have added the following new screens:
```
screens/BookingFormScreen.tsx
containers/Form/BookingForm.tsx
```
To accommodate the needs of these screens and to centralize the functions used in the booking service, I have created the helper file `helpers/BookingHelper.ts` for helper functions related to booking, and `types/BookingTypes.ts` to contain the types used in the booking system.
In addition, I have updated several existing files to make use of the new types, and solved a few minor bugs related to the behaviour of the components used in the new booking form screens.

## How to test the changes?

First make sure the backend is set up correctly:
1. Deploy the latest version of `booking`, `bookables` and `timeslot` services from the `sls-api` and `sls-resources` respositories.
2. Deploy the correct `datatorgetEnvs` and `bookablesEnvs` parameters to the backend.
3. Make sure you have access to the environment that datatorget is connected to, and that you have a user in a shared mailbox group that is defined in one of the services described in `bookablesEnvs`.
4. Create a booking in the Outlook calendar of the user with the exact title "MH bokningsbar tid".

Now you can launch the app to test the changes:
1. Checkout this branch
2. Make sure the env is not in storybook mode
4. Fire upp the simulator by running the command `yarn ios`
5. Navigate to the calendar screen
6. Test the new functionalities by pressing "Boka möte"

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [X] Building the Application on a iOS device/simulator.
- [X] Building the Application on a Android device/simulator.

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 22 Issues 
* Code issue 22 Code issues 

 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/fedfce3fd8695f238dea9cb00bd37ce7/pull-request/574)

File: source/components/molecules/BottomModal/styled.ts [hunk 1]
```diff
@@ -14,15 +15,38 @@ interface ModalTopContainerProps {
...
+  width: 32px;
+  height: 5px;
+  margin-top: 8px;
+  margin-bottom: 0px;
+  border-radius: 55px;
+`;
+
+const InnerNavigatorContainer = styled.View<ModalTopContainerProps>`
+  display: flex;
+  flex-direction: row;
```
Comment (karlknowit) [inline source/components/molecules/BottomModal/styled.ts:41]: flex-direction : row is default in flex container, only needed if the parent has a flex-direction that is not row

File: source/components/molecules/DateTimeCard/styled.ts [hunk 2]
```diff
@@ -1,33 +1,42 @@
...
+const Container = styled.View<{ align: string }>`
   margin-top: 10px;
   border-width: 3px;
   border-radius: 5px;
   padding: 10px;
   display: flex;
   flex-direction: column;
   justify-content: center;
-  align-items: center;
+  align-items: ${(props) => props.align};
```
Comment (karlknowit) [inline source/components/molecules/DateTimeCard/styled.ts:11]: {( { align }) => align}

File: source/screens/bookingScreens/CalendarScreen.tsx [hunk 3]
```diff
@@ -32,6 +37,30 @@ const ListHeading = styled(Heading)`
   margin-bottom: 8px;
 `;
 
+const ScrollViewSpacer = styled.View`
```
Comment (karlknowit) [inline source/screens/bookingScreens/CalendarScreen.tsx:40]: OPTIONAL: 

Move styling to its own file

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/components/molecules/CaseCard/CaseCard.tsx [hunk 4]
```diff
@@ -37,6 +37,7 @@ interface CaseCardProps {
   showBookingDate?: boolean;
   bookingDate?: string;
   bookingTime?: string;
+  smallDateTimeCard?: boolean;
```
Comment (karlknowit) [inline source/components/molecules/CaseCard/CaseCard.tsx:40]: how about changing the property to `dateTimeCardSize` which is a string of with value "small" or "large" ?

File: source/containers/Form/BookingForm.tsx [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+import { CharacterCard, TimeSlotPicker } from "../../components/molecules";
+import { Button, Text } from "../../components/atoms";
+import FormField from "../FormField/FormField";
+import { validateInput } from "../../helpers/ValidationHelper";
+
+const Scroller = styled(KeyboardAwareScrollView)`
+  flex: 1;
+`;
+
+const ListWrapper = styled.View`
```
Comment (karlknowit) [inline source/containers/Form/BookingForm.tsx:20]: OPTIONAL:

Style in separate file perhaps?

File: source/containers/Form/BookingForm.tsx [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+  isContactsMode,
+  submitPending,
+  onSubmit,
+}: BookingFormProps): JSX.Element => {
+  const [answers, setAnswers] = useState<Record<string, string>>({});
+  const [validationErrors, setValidationErrors] = useState<
+    Record<string, ValidationError>
+  >({});
+  const [timeSlot, setTimeSlot] = useState<TimeSlot | undefined>();
+  const [currentEmail, setCurrentEmail] = useState<string>("");
```
Comment (karlknowit) [inline source/containers/Form/BookingForm.tsx:68]: the explicit type <string> is not needed

**Review by karlknowit — CHANGES_REQUESTED** [review]

File: source/screens/bookingScreens/CalendarScreen.tsx [hunk 7]
```diff
@@ -32,6 +37,30 @@ const ListHeading = styled(Heading)`
...
+  height: 60px;
+`;
+
+const SmallCard = styled.View`
+  display: flex;
+  flex-direction: row;
```
[the remark this answers is not in the document]
Comment (sepehrnoor) [reply source/screens/bookingScreens/CalendarScreen.tsx:46]: Personally I'd like to keep flex-direction for the sake of clarity.

**Review by karlknowit — APPROVED** [review]

**Pull request merged**